libexec/rtld-elf/rtld_c18n.c

```diff
@@ -1419,22 +1471,26 @@ c18n_init(void)
 
 	atomic_store_explicit(&tramp_pgs.head, tramp_pg_new(NULL),
 	    memory_order_relaxed);
-}
 
-void *
-c18n_return_address(void)
-{
-	struct trusted_frame *tframe;
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wframe-address"
 	/*
-	 * Unwind twice to locate the trusted frame.
+	 * Wrap the hooks used by external libraries in trampolines.
 	 */
-	tframe = __builtin_frame_address(2);
-#pragma clang diagnostic pop
-
-	return (tframe->pc);
+	_rtld_sighandler = tramp_intern(NULL, &(struct tramp_data) {
+		.target = _rtld_sighandler_unsafe,
+		.defobj = &obj_rtld,
```

Comment:
```suggestion
		.defobj = obj_rtld,
```

Would seem to be needed now as well (and below)?